crates/ruff/src/rules/pyflakes/fixes.rs

```diff
@@ -90,31 +90,28 @@ pub(crate) fn remove_unused_positional_arguments_from_format_call(
 
 /// Generate a [`Edit`] to remove the binding from an exception handler.
 pub(crate) fn remove_exception_handler_assignment(
-    except_handler: &ExceptHandler,
+    range: TextRange,
     locator: &Locator,
 ) -> Result<Edit> {
-    let contents = locator.slice(except_handler.range());
-    let mut fix_start = None;
-    let mut fix_end = None;
-
-    // End of the token just before the `as` to the semicolon.
-    let mut prev = None;
-    for (tok, range) in
-        lexer::lex_starts_at(contents, Mode::Module, except_handler.start()).flatten()
-    {
-        if tok.is_as() {
-            fix_start = prev;
-        }
-        if tok.is_colon() {
-            fix_end = Some(range.start());
-            break;
-        }
-        prev = Some(range.end());
-    }
-
-    if let (Some(start), Some(end)) = (fix_start, fix_end) {
-        Ok(Edit::deletion(start, end))
-    } else {
-        bail!("Could not find span of exception handler")
-    }
+    // Lex backwards, to the token just before the `as`.
+    let mut tokenizer = SimpleTokenizer::up_to(range.start(), locator.contents()).skip_trivia();
+
+    // Eat the `as` token.
+    let preceding = tokenizer
+        .next_back()
+        .context("expected the exception name to be preceded by `as`")?;
+    debug_assert!(matches!(preceding.kind, TokenKind::As));
+
+    // Lex to the end of the preceding token, which should be the exception value.
+    let preceding = tokenizer
+        .next_back()
+        .context("expected the exception name to be preceded by a token")?;
```

Comment:
What are the valid tokens for the exception value? The simple tokenizer (intentionally) only supports a very limited set of tokens. E.g. strings no strings or identifiers.

Reply:
We just want _anything_ that isn't a continuation.

Reply:
It's basically `X as Y`, and we're starting at `Y`. The only thing you can have between `X` and `as`, and between `as` and `Y`, is whitespace and continuation (backslash).